File: solutions/3237. Alt and Tab Simulation/3237.py
class Solution:
  def simulationResult(
      self,
      windows: list[int],
      queries: list[int],
  ) -> list[int]:
    ans = []
    seen = set()

    for query in reversed(queries):
      if query not in seen:
        ans.append(query)
        seen.add(query)

    for window in windows:
      if window not in seen:
        ans.append(window)
        seen.add(window)

    return ans

File: solutions/3237. Alt and Tab Simulation/3237.cpp
class Solution {
 public:
  vector<int> simulationResult(vector<int>& windows, vector<int>& queries) {
    vector<int> ans;
    unordered_set<int> seen;

    for (int i = queries.size() - 1; i >= 0; --i)
      if (!seen.contains(queries[i])) {
        ans.push_back(queries[i]);
        seen.insert(queries[i]);
      }

    for (const int window : windows)
      if (!seen.contains(window)) {
        ans.push_back(window);
        seen.insert(window);
      }

    return ans;
  }
};
